client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -119,12 +163,22 @@ public ReaderGroupConfig build() {
            checkArgument(startingStreamCuts != null && startingStreamCuts.size() > 0,
                    "Stream names that the reader group can read from cannot be empty");
 
-           startingStreamCuts.forEach((s, streamCut) -> {
+           //endStreamCuts is an optional configuration. Initialize if it is null.
+           if (endingStreamCuts == null) {
+               endingStreamCuts = Collections.emptyMap();
+           }
+
+           Consumer<Map<Stream, StreamCut>> validateStreamCuts = map -> map.forEach((s, streamCut) -> {
```

Comment:
This is a lot of overhead to avoid simply creating a private method...

Reply:
:-) Yes, added a private method.